Add support for adding commits to a repository (#116)
pulp/pulp_ostree: docs/workflows/upload.rst

    }

Upload a Tarball
----------------

> At the beginning, create a local OSTree repository composed of one commit. Initialize the repository

Reviewer: ```suggestion
First, create a local OSTree repository composed of one commit. Initialize the repository
```

and commit one of the files, e.g., published at `www.gnu.org <https://ftp.gnu.org/gnu/bash/>`_::

    ostree --repo=repo init --mode=archive
    ostree --repo=repo commit -b stable --tree=tar=bash-doc-3.2.tar.gz

> Now, upload the repository to Pulp as an artifact. Each artifact in Pulp represents a file.

Reviewer: ```suggestion
Then, upload the repository to Pulp as an artifact. Each artifact in Pulp represents a file.
```

or just "Upload"

        ...
    }

Finally, create a new repository version by submitting the artifact which describes the OSTree
repository::

    http POST ${BASE_ADDR}${REPO_HREF}upload/ artifact=${ARTIFACT_HREF} repository_name=repo

Response::

    {
        "task": "/pulp/api/v3/tasks/5e93290c-c60c-478b-abf2-bbadb6d1d9e8/"
    }

Add Commits to the Repository
-----------------------------

If there is a need to upload additional commits to an existing repository, one can do so by using
the endpoint for uploading tarballs again::

    mkdir tree && echo "Hello World!" > tree/hello.txt
    ostree --repo=repo-next init --mode=archive
    ostree --repo=repo-next commit -b stable tree/

    tar -cvf repo-next.tar repo-next
    pulp artifact upload --file repo-next.tar

Response::

    {
        ...
        /pulp/api/v3/artifacts/5b684f15-f10d-4e80-8ea0-73dd98c61fc6/
        ...
    }

Upload the commit::

    http POST ${BASE_ADDR}${REPO_HREF}upload/ artifact=${ARTIFACT_HREF} repository_name=repo ref=stable parent_commit=528ac3ca991bdd42d07c32761639cff6cc203e9b7135605081ffdbac7ddb1191

Response::

    {
        "task": "/pulp/api/v3/tasks/f3259790-7758-4760-8d75-41008ef32e3d/"
    }

> The commit which introduced the file ``hello.txt`` was added to the repository and attached to the

Reviewer: ```suggestion
The commit that introduced the file ``hello.txt`` was added to the repository and attached to the
```
